ray-operator/controllers/ray/raycluster_controller.go

```diff
@@ -83,21 +83,96 @@ type RayClusterReconciler struct {
 // +kubebuilder:rbac:groups="rbac.authorization.k8s.io",resources=rolebindings,verbs=get;list;watch;create;delete
 // Reconcile used to bridge the desired state with the current state
 func (r *RayClusterReconciler) Reconcile(ctx context.Context, request ctrl.Request) (ctrl.Result, error) {
-	_ = r.Log.WithValues("raycluster", request.NamespacedName)
-	log.Info("reconciling RayCluster", "cluster name", request.Name)
+	var err error
+
+	// Try to fetch the Event instance
+	event := &v1.Event{}
+	if err = r.Get(context.TODO(), request.NamespacedName, event); err == nil {
+		return r.eventReconcile(request, event)
+	}
 
-	// Fetch the RayCluster instance
+	// Try to fetch the RayCluster instance
 	instance := &rayiov1alpha1.RayCluster{}
-	if err := r.Get(context.TODO(), request.NamespacedName, instance); err != nil {
-		if errors.IsNotFound(err) {
-			log.Info("Read request instance not found error!")
-		} else {
-			log.Error(err, "Read request instance error!")
+	if err = r.Get(context.TODO(), request.NamespacedName, instance); err == nil {
+		return r.rayClusterReconcile(request, instance)
+	}
+
+	// No match found
+	if errors.IsNotFound(err) {
+		log.Info("Read request instance not found error!", "name", request.NamespacedName)
+	} else {
+		log.Error(err, "Read request instance error!")
+	}
+	// Error reading the object - requeue the request.
+	return ctrl.Result{}, client.IgnoreNotFound(err)
+}
+
+func (r *RayClusterReconciler) eventReconcile(request ctrl.Request, event *v1.Event) (ctrl.Result, error) {
```

Comment:
Do we need `eventReconcile`? What does this do? I see the logic is to do logs and add annotation?

Reply:
eventReconcile is for detecting readiness probe failures.